cpp/daal/src/algorithms/k_nearest_neighbors/bf_knn_impl.i

```diff
@@ -72,29 +75,32 @@ public:
         NumericTable * newTrainLabelTable = const_cast<NumericTable *>(trainLabelTable);
         if (resultsToEvaluate & daal::algorithms::classifier::computeClassLabels)
         {
-            newTrainLabelTable->getBlockOfRows(0, nTrain, readWrite, trainLabelBlock);
+            newTrainLabelTable->getBlockOfRows(0, nTrain, readOnly, trainLabelBlock);
             trainLabel = trainLabelBlock.getBlockPtr();
             DAAL_CHECK_MALLOC(trainLabel);
         }
 
-        services::SharedPtr<daal::algorithms::internal::PairwiseDistances<FPType, cpu> > dist;
+        services::SharedPtr<PairwiseDistances<FPType, cpu> > dist;
 
-        switch (pairwiseDistance)
+        if (pairwiseDistance == PairwiseDistanceType::minkowski && minkowskiDegree == 2.0)
```

Comment:
Switch is preferred here (with 'if' in minkowski case)

Reply:
How was it better before?